packages/dds/tree/src/simple-tree/schemaFactoryRecursive.ts

```diff
@@ -192,11 +40,57 @@ export function createFieldSchemaUnsafe<
 }
 
 /**
- * Extends SchemaFactory with utilities for recursive types.
+ * Extends SchemaFactory with utilities for recursive schema.
+ *
+ * @remarks
+ * This is separated from SchemaFactory as these APIs are more experimental, and may be stabilized independently.
+ *
+ * The non-recursive versions of the schema building methods will run into several issues when used recursively.
+ * Consider the following example:
  *
- * @remarks This is separated from SchemaFactory as these APIs are more experimental, and may be stabilized independently.
+ * ```typescript
+ * const Test = sf.array(Test); // Bad
+ * ```
  *
- * @sealed @internal
+ * This has several issues:
+ *
+ * 1. It is a structurally named schema.
+ * Structurally named schema derive their name from the names of their child types, which is not possible when the type is recursive since its name would include itself.
+ * Instead a name must be explicitly provided.
+ *
+ * 2. The schema accesses itself before its defined.
+ * This would be a runtime error if the TypeScript compiler allowed it.
+ * This can be fixed by wrapping the type in a function, which also requires explicitly listing the allowed types an an array (`[() => Test]`)
```

Comment:
Nit:
```suggestion
 * This can be fixed by wrapping the type in a function, which also requires explicitly listing the allowed types in an array (`[() => Test]`).
```